SpikesBot.js

```diff
@@ -0,0 +1,311 @@
+    });
+    // set a timeout for the recursive function that is sending the teamtime every day
+    if(teamTime !== null){
+        setTimeout(() => {sendTeamTime(getAllGeneralChannels())}, whenIsTeamTime())
+    }
+
+});
+
+// recursive function that is sending the teamtime every day at the correct time
+function sendTeamTime (channels){
+    const now = new Date() //the current time
+    
+    //for each general channel from the list
+    channels.forEach((channel) =>{
+
+        //if it is the special date (example: for the team TheSpikes#2212 it is dec 22 or 22/12)
+        if(now.getDate() == teamTime.getHours() && now.getMonth()+1 === teamTime.getMinutes()){
+            //send teamascii
+            channel.send(teamAscii)
+        }
+        else{
+            //send teamtime
+            //channel.send(teamTime.getHours() + '' + teamTime.getMinutes())
+            channel.send(`${teamTime.getHours().toString().padStart(2, '0')}:${teamTime.getMinutes().toString().padStart(2, '0')}`)
+        }
+    })
+
+    // after sending, wait a minuate before timing the next messeges
+    const waitAMinuate = () => {        
+        setTimeout(() => {sendTeamTime(getAllGeneralChannels())}, whenIsTeamTime())
+    }
+    setTimeout(waitAMinuate, 60000)
+}
+
+// a function for getting the new and up to date list of general channels from each server
+function getAllGeneralChannels(){
+    let generalChannels = []
+
+    //for each server
+    client.guilds.cache.forEach((guild) => {
+        //for each channel
+        guild.channels.cache.forEach((guildChannel) =>{            
+
+            //if the channel named general
+            if(guildChannel.name === 'general'){
+                //add it to the list
+                generalChannels[generalChannels.length] = guildChannel
+            }
+        })
+    });
+    return generalChannels;
+}
+
+//calc when to send the teamTime message
+function whenIsTeamTime(){
+    const now = new Date() //the current time
+
+    //calc the remaing time until 22:12
+    let remain = (teamTime.getHours() - now.getHours()) * 60 * 60 * 1000
+    remain += (teamTime.getMinutes() - now.getMinutes()) * 60 * 1000
+    remain -= now.getSeconds() * 1000
+
+    if(remain >= 0){
+    }
+    else{
+        remain += 24 * 60 * 60 * 1000
+    }
+
+    //return the time until the next 22:12
+    console.log("next teamTime: " + remain);
+    return remain;
+}
+
+//fetch info about a team from the blue alliance website
+function fetchTeamLinks(someTeamNumber) {
+
+    return fetch('https://www.thebluealliance.com/team/' + someTeamNumber)
+        .then(response => response.text())
+        .then(htmlContent => {
+            //get the html area of the team's social media accounts
+            let social = htmlContent.split('team-social-media')[1].split('</div>')[0]            
+            let links = social.split('href="')
+            
+            //for each link
+            for(let i = 0; i < links.length; i++){
```

Comment:
`for (const item of iterable)`

Reply:
it wouldn't work in that case because JavaScript